### PR TITLE
Changes renaming habitat-api to habitat-lab

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
We changed the name of Habitat-API to Habitat Lab to make less confusing and stress the purpose of the repo to propose Tasks, Metrics, Baselines to the community.

## How Has This Been Tested
CI + visiting the branch:
Screen Shot 2020-08-12 at 2 22 53 PM

## Types of changes

- Docs change / refactoring / dependency upgrade
- Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: docs/pages/habitat-api-demo.rst [hunk 1]
```diff
@@ -1,11 +1,11 @@
 Habitat API Demo
 ################
 
-.. button-primary:: https://dl.fbaipublicfiles.com/habitat/notebooks/habitat-api-demo.ipynb
```
Comment (Skylion007) [inline docs/pages/habitat-api-demo.rst:4]: Hold on! Was this file renamed in the URL? @mathfac

**Comment by Skylion007** [comment]
@mathfac Is there any automated way we can check if the URLs in the repo are valid? Like with a pre-commit hook?

File: docs/pages/habitat-api-demo.rst [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+.. button-primary:: https://dl.fbaipublicfiles.com/habitat/notebooks/habitat-lab-demo.ipynb
```
Comment (Skylion007) [inline docs/pages/habitat-api-demo.rst:4]: @mathfac Confirmed URL is invalid

File: docs/pages/habitat-api-demo.rst [hunk 3]
```diff
@@ -1,11 +1,11 @@
 Habitat API Demo
```
Comment (Skylion007) [inline docs/pages/habitat-api-demo.rst:1]: This RST has to be renamed too. You need a sed in filenames too @mathfac

File: docs/conf.py [hunk 4]
```diff
@@ -35,7 +35,7 @@
     "pages/index.rst",
     "pages/quickstart.rst",
     "pages/habitat-sim-demo.rst",
-    "pages/habitat-api-demo.rst",
```
Comment (Skylion007) [inline docs/conf.py:38]: Invalid unless you rename it.

File: docs/build.sh [hunk 5]
```diff
@@ -35,4 +35,4 @@ $mcss_path/documentation/python.py conf.py
 echo "------------------------------------------------------------------------"
 echo "Docs were successfully generated. Open the following link to view them:"
 echo
-echo "file://$(pwd)/../../habitat-sim/build/docs/habitat-api/index.html"
```
Comment (Skylion007) [inline docs/build.sh:38]: I don't see any formative changes in the build script though? I don't think these urls will be valid...

File: docs/pages/habitat-api-demo.rst [hunk 6]
```diff
@@ -1,12 +1,6 @@
...
 ################
 
-.. button-primary:: https://dl.fbaipublicfiles.com/habitat/notebooks/habitat-lab-demo.ipynb
```
Comment (Skylion007) [inline docs/pages/habitat-api-demo.rst]: Why? Just rename and reupload the ipynb. We can convert it to a colab as well.

File: docs/build.sh (hunk 5 above)
Comment (mathfac) [reply docs/build.sh:38]: There will be separate PR for HSIM and HWebsite and it became valid.

**Comment by mathfac** [comment]
@Skylion007 the documentation built successfully. We should be good to go.

**Pull request merged**